core/lib/circuit_breaker/src/lib.rs

```diff
@@ -21,48 +33,42 @@ pub enum CircuitBreakerError {
 /// Checks circuit breakers
 #[derive(Debug)]
 pub struct CircuitBreakerChecker {
-    circuit_breakers: Vec<Box<dyn CircuitBreaker>>,
+    circuit_breakers: Arc<CircuitBreakers>,
     sync_interval: Duration,
 }
 
 #[async_trait::async_trait]
 pub trait CircuitBreaker: std::fmt::Debug + Send + Sync {
+    fn name(&self) -> &'static str;
+
     async fn check(&self) -> Result<(), CircuitBreakerError>;
 }
 
 impl CircuitBreakerChecker {
-    pub fn new(
-        circuit_breakers: Vec<Box<dyn CircuitBreaker>>,
-        config: &CircuitBreakerConfig,
-    ) -> Self {
+    pub fn new(circuit_breakers: Arc<CircuitBreakers>, sync_interval: Duration) -> Self {
         Self {
             circuit_breakers,
-            sync_interval: config.sync_interval(),
+            sync_interval,
         }
     }
 
     pub async fn check(&self) -> Result<(), CircuitBreakerError> {
-        for circuit_breaker in &self.circuit_breakers {
+        for circuit_breaker in self.circuit_breakers.0.lock().await.iter() {
             circuit_breaker.check().await?;
         }
         Ok(())
     }
```

Comment:
You can move this method to `CircuitBreakers` -- it probably logically belongs there (and it'll be consistent with the healthcheck module design).